packages/netlify-cms-core/src/components/Editor/Editor.js

```diff
@@ -364,6 +364,7 @@ class Editor extends React.Component {
       logoutUser,
       deployPreview,
       loadDeployPreview,
+      draftKey,
       slug,
```

Comment:
Added a unique id to serve as a key when a new draft is loaded - this allows us to reliably reset the entire Editor component when we're dealing with a different entry to avoid weird nested state issues.